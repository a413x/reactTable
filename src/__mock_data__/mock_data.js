export const mock_data = [
  {
    "id":"1",
    "firstName":"Frederick",
    "lastName":"Catania",
    "email":"GHellems@magna.net",
    "phone":"(458)969-6847",
    "address":{"streetAddress":"7471 Porttitor Ln","city":"Gig Harbor","state":"WY","zip":"15219"},
    "description":"sit pretium ipsum consectetur eros"
  },
  {
    "id":"2",
    "firstName":"Curtis",
    "lastName":"Bias",
    "email":"CRobichaud@nec.gov",
    "phone":"(481)075-5034",
    "address":{"streetAddress":"2730 Risus Ln","city":"West Warwick","state":"NM","zip":"60799"},
    "description":"sollicitudin malesuada tellus porta neque"
  },
  {
    "id":"3",
    "firstName":"Dora",
    "lastName":"Crenshaw",
    "email":"MSanchez@at.org",
    "phone":"(959)257-7541",
    "address":{"streetAddress":"5670 At Ave","city":"Pleasant Grove","state":"PA","zip":"21878"},
    "description":"amet lacus et fringilla orci"
  },
  {
    "id":"4",
    "firstName":"Grace",
    "lastName":"French",
    "email":"JLund@vestibulum.com",
    "phone":"(976)555-7134",
    "address":{"streetAddress":"1933 Vestibulum Rd","city":"Woonsocket","state":"FL","zip":"39433"},
    "description":"velit sed tellus suspendisse at"
  },
  {
    "id":"5",
    "firstName":"Randy",
    "lastName":"Bunch",
    "email":"KGunter@lacus.io",
    "phone":"(941)539-5519"
  }
]
