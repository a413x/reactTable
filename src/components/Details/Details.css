.details{
  position: relative;
}
.details .close{
  position: absolute;
  top: 5px;
  right: 10px;
}
