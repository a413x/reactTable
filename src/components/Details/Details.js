import './Details.css'

export const Details = ({dataObj, closeDetailsCallback}) => {
  const {firstName, lastName, description, address} = dataObj
  return(
    <div className = 'details card my-3 d-inline-block'>
      <div className = 'card-body'>
        <p>User selected: <b>{firstName + ' ' + lastName}</b></p>
        {description && <p className = 'd-flex'>
          <span className='mr-3'>Description:</span>
          <textarea value = {description} className = 'form-control' readOnly>
          </textarea>
        </p>}

        {address && <div>
          <p>Address: <b>{address.streetAddress}</b></p>
          <p>City: <b>{address.city}</b></p>
          <p>State: <b>{address.state}</b></p>
          <p>Index: <b>{address.zip}</b></p>
        </div>}

        <button type = 'button' className = 'close' aria-label = 'Close'
          onClick = {closeDetailsCallback}>
          <span aria-hidden = 'true'>&times;</span>
        </button>
      </div>
    </div>
  )
}
