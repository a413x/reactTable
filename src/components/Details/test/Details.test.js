import {render, fireEvent} from '@testing-library/react'
import {Details} from '../Details.js'
import {mock_data} from '../../test/mock_data.js'

describe('Details component tests', () => {
  const mock_closeDetailsCallback = jest.fn()
  const props = {
    dataObj: {...mock_data[2]},
    closeDetailsCallback: mock_closeDetailsCallback
  }
  test('renders properly', () => {
    const {container} = render(<Details {...props} />)
    expect(container).toMatchSnapshot()
  })
  test('renders properly without address and description', () => {
    props.dataObj = {...mock_data[4]}
    const {container} = render(<Details {...props} />)
    expect(container).toMatchSnapshot()
  })
  test('Close button calls close callback function', () => {
    render(<Details {...props} />)
    const button = document.querySelector('.close')
    fireEvent.click(button)
    expect(mock_closeDetailsCallback).toHaveBeenCalledTimes(1)
  })
})
