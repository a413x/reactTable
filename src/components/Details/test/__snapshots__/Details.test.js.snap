// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Details component tests renders properly 1`] = `
<div>
  <div
    class="details card my-3 d-inline-block"
  >
    <div
      class="card-body"
    >
      <p>
        User selected: 
        <b>
          Dora Crenshaw
        </b>
      </p>
      <p
        class="d-flex"
      >
        <span
          class="mr-3"
        >
          Description:
        </span>
        <textarea
          class="form-control"
          readonly=""
        >
          amet lacus et fringilla orci
        </textarea>
      </p>
      <div>
        <p>
          Address: 
          <b>
            5670 At Ave
          </b>
        </p>
        <p>
          City: 
          <b>
            Pleasant Grove
          </b>
        </p>
        <p>
          State: 
          <b>
            PA
          </b>
        </p>
        <p>
          Index: 
          <b>
            21878
          </b>
        </p>
      </div>
      <button
        aria-label="Close"
        class="close"
        type="button"
      >
        <span
          aria-hidden="true"
        >
          ×
        </span>
      </button>
    </div>
  </div>
</div>
`;

exports[`Details component tests renders properly without address and description 1`] = `
<div>
  <div
    class="details card my-3 d-inline-block"
  >
    <div
      class="card-body"
    >
      <p>
        User selected: 
        <b>
          Randy Bunch
        </b>
      </p>
      <button
        aria-label="Close"
        class="close"
        type="button"
      >
        <span
          aria-hidden="true"
        >
          ×
        </span>
      </button>
    </div>
  </div>
</div>
`;
